src/WindowsTracing/GraphicsEtwProvider.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WINDOWS_TRACING_GRAPHICS_ETW_PROVIDER_H_
+#define WINDOWS_TRACING_GRAPHICS_ETW_PROVIDER_H_
+
+#include <absl/container/flat_hash_map.h>
+
+#include <krabs/krabs.hpp>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadConstants.h"
+#include "WindowsTracing/TracerListener.h"
+
+namespace orbit_windows_tracing {
+
+// This class sets up a krabs::user_trace so that it receives graphics related ETW events. Those
+// events are converted into PresentEvent objects that are relayed to a TraceListener.
+class GraphicsEtwProvider {
+ public:
+  GraphicsEtwProvider(uint32_t pid, krabs::user_trace* user_trace, TracerListener* listener);
+
+  void OutputStats();
+
+ private:
+  void EnableProvider(std::string_view name, GUID guid, krabs::provider_event_callback callback);
```

Comment:
I suggest to change the signature of the function. IMO it should rather take a pointer to a member finction and create an instance of `krabs::provider_event_callback` itself. This will save you the need to manually wrap a method ~10 times in `GraphicsEtwProvider::GraphicsEtwProvider`.

Reply:
Hmm, I think `absl::bind_front()` is concise enough, and the code block formats well, I'll leave it as is.